hack/lib/lint.sh

```diff
@@ -20,31 +20,36 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-SED_CMD=""
-KUBEEDGE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)"
+function kubeedge::lint::init() {
+    SED_CMD=""
+    KUBEEDGE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)"
 
-if [[ "$OSTYPE" == "darwin"* ]]
-then
-    SED_CMD=`which gsed`
-    if [ -z $SED_CMD ]
+    if [[ "$OSTYPE" == "darwin"* ]]
     then
-        echo "Please install gnu-sed (brew install gnu-sed)"
-        exit 1
-    fi
-elif [[ "$OSTYPE" == "linux"* ]]
-then
-    SED_CMD=`which sed`
-    if [ -z $SED_CMD ]
+        if ! which gsed >/dev/null 2>&1
```

Comment:
I don't think this is necessary. IMHO, these two are equal :)
```
SED_CMD=`which gsed`
    if [ -z $SED_CMD ]
```
`if ! which gsed >/dev/null 2>&1`

What is the output of  `which gsed` or `make lint` cmd if gsed is not installed?

Reply:
you can try it in your linux host, modify the sed to as `sedddd`, seems needed.

Reply:
> I don't think this is necessary. IMHO, these two are equal :)
> 
> ```
> SED_CMD=`which gsed`
>     if [ -z $SED_CMD ]
> ```
> 
> `if ! which gsed >/dev/null 2>&1`
> 
> What is the output of `which gsed` or `make lint` cmd if gsed is not installed?

Hi, it will output this in master:
![image](https://user-images.githubusercontent.com/62384022/149893871-01d420d8-4051-49f7-8b3e-a9973ad61c33.png)

Reply:
oh, i see, thanks for fixing

Reply:
I guess it's due to `` mark
```
SED_CMD=`which sed`
    if [ -z $SED_CMD ]
```
```
SED_CMD="which sed"
    if [ -z "$SED_CMD" ]
```

Reply:
> you can try it in your linux host, modify the sed to as `sedddd`, seems needed.

I try it in my ubuntu:

![image](https://user-images.githubusercontent.com/62384022/149895752-ec26dbe7-9b02-4938-98e0-912eb30e4b0d.png)

![image](https://user-images.githubusercontent.com/62384022/149895577-5af8f215-812a-4e63-b6db-1f2c0ecae335.png)

Reply:
if we want to get the result of cmd running using ``, we must ensure the cmd can be running successfully, so in the before way, the script exist unnormally when running `which sed` cmd.